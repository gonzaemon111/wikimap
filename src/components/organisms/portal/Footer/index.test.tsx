import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import * as Stories from './index.stories';

describe('Organisms/Footer コンポーネント', () => {
  test('適切なclassNameが適用されている', () => {
    const { Default } = composeStories(Stories);
    const { getByText } = render(<Default />);
    expect(getByText('サービス')).toBeTruthy();
  });

  describe('スナップショットテスト', () => {
    const testCases = Object.values(composeStories(Stories)).map((Story) => [Story.storyName, Story]);

    test.each(testCases)('Renders %s story', async (storyName, Story) => {
      if (!Story) {
        throw new Error('Story not Found');
      }

      const tree = render(<Story />);
      expect(tree).toMatchSnapshot();
    });
  });
});
