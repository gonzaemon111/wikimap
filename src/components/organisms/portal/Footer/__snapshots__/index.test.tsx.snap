// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/Footer コンポーネント スナップショットテスト Renders Default story 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="footer bg-neutral p-10 text-neutral-content"
      >
        <nav>
          <header
            class="footer-title"
          >
            サービス
          </header>
          <a
            class="link-hover link"
          >
            Branding
          </a>
          <a
            class="link-hover link"
          >
            Design
          </a>
          <a
            class="link-hover link"
          >
            Marketing
          </a>
          <a
            class="link-hover link"
          >
            Advertisement
          </a>
        </nav>
        <nav>
          <header
            class="footer-title"
          >
            Company
          </header>
          <a
            class="link-hover link"
          >
            About us
          </a>
          <a
            class="link-hover link"
          >
            Contact
          </a>
          <a
            class="link-hover link"
          >
            Jobs
          </a>
          <a
            class="link-hover link"
          >
            Press kit
          </a>
        </nav>
        <nav>
          <header
            class="footer-title"
          >
            Legal
          </header>
          <a
            class="link-hover link"
          >
            Terms of use
          </a>
          <a
            class="link-hover link"
          >
            Privacy policy
          </a>
          <a
            class="link-hover link"
          >
            Cookie policy
          </a>
        </nav>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="footer bg-neutral p-10 text-neutral-content"
    >
      <nav>
        <header
          class="footer-title"
        >
          サービス
        </header>
        <a
          class="link-hover link"
        >
          Branding
        </a>
        <a
          class="link-hover link"
        >
          Design
        </a>
        <a
          class="link-hover link"
        >
          Marketing
        </a>
        <a
          class="link-hover link"
        >
          Advertisement
        </a>
      </nav>
      <nav>
        <header
          class="footer-title"
        >
          Company
        </header>
        <a
          class="link-hover link"
        >
          About us
        </a>
        <a
          class="link-hover link"
        >
          Contact
        </a>
        <a
          class="link-hover link"
        >
          Jobs
        </a>
        <a
          class="link-hover link"
        >
          Press kit
        </a>
      </nav>
      <nav>
        <header
          class="footer-title"
        >
          Legal
        </header>
        <a
          class="link-hover link"
        >
          Terms of use
        </a>
        <a
          class="link-hover link"
        >
          Privacy policy
        </a>
        <a
          class="link-hover link"
        >
          Cookie policy
        </a>
      </nav>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
